src/EditorFeatures/CSharpTest2/Recommendations/GlobalKeywordRecommenderTests.cs

```diff
@@ -170,5 +170,145 @@ class C
+        public async Task TestBeforeUsing()
+        {
+            await VerifyKeywordAsync(
+@"$$
+using Goo;");
+        }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.KeywordRecommending)]
+        public async Task TestBeforeGlobalUsing()
+        {
+            await VerifyKeywordAsync(
+@"$$
+global using Goo;");
+        }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.KeywordRecommending)]
+        public async Task TestAfterUsingAlias()
+        {
+            await VerifyKeywordAsync(
+@"using Goo = Bar;
+$$");
+        }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.KeywordRecommending)]
+        public async Task TestAfterGlobalUsingAlias()
+        {
+            await VerifyKeywordAsync(
+@"using Goo = Bar;
+$$");
+        }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.KeywordRecommending)]
+        public async Task TestNotAfterGlobalKeyword()
+        {
+            await VerifyAbsenceAsync(@"
+global $$
+");
+        }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.KeywordRecommending)]
+        public async Task TestNotAfterUsingKeyword()
+        {
+            await VerifyAbsenceAsync(@"
+using $$
+");
+        }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.KeywordRecommending)]
+        public async Task TestNotAfterGlobalUsingKeyword()
+        {
+            await VerifyAbsenceAsync(@"
+global using $$
+");
+        }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.KeywordRecommending)]
+        public async Task TestNotBeforeExtern()
+        {
+            await VerifyAbsenceAsync(SourceCodeKind.Regular,
+@"$$
+extern alias Goo;");
+        }
+
+        [Fact(Skip = "https://github.com/dotnet/roslyn/issues/9880"), Trait(Traits.Feature, Traits.Features.KeywordRecommending)]
+        public async Task TestNotBeforeExtern_Interactive()
+        {
+            await VerifyAbsenceAsync(SourceCodeKind.Script,
+@"$$
+extern alias Goo;");
+        }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.KeywordRecommending)]
+        public async Task TestBetweenGlobalUsings_01()
+        {
+            await VerifyKeywordAsync(
+@"global using Goo;
+$$
+global using Bar;");
+        }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.KeywordRecommending)]
+        public async Task TestBetweenUsings_02()
+        {
+            await VerifyKeywordAsync(
+@"global using Goo;
+$$
+using Bar;");
+        }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.KeywordRecommending)]
+        public async Task TestNotAfterGlobalBetweenGlobalUsings_01()
+        {
+            await VerifyAbsenceAsync(
+@"global using Goo;
+global $$
+global using Bar;");
+        }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.KeywordRecommending)]
+        public async Task TestNotAfterGlobalBetweenUsings_02()
+        {
+            await VerifyAbsenceAsync(
+@"global using Goo;
+global $$
+using Bar;");
```

Comment:
can you add a test about adding it before a namespace, class, top level statement, and global/normal attributes?